src/components/AddTodo.tsx

```diff
@@ -21,82 +31,144 @@ import {
+                    <CalendarIcon className="mr-2 h-4 w-4" />
+                    {formik.values.date
+                      ? format(new Date(formik.values.date), "PPP")
+                      : "Select a date"}
                   </Button>
                 </PopoverTrigger>
                 <PopoverContent className="w-auto p-0" align="start">
                   <Calendar
                     mode="single"
-                    selected={date}
+                    selected={
+                      formik.values.date
+                        ? new Date(formik.values.date)
+                        : undefined
+                    }
                     onSelect={(date) => {
-                      setDate(date || undefined);
+                      formik.setFieldValue("date", date || undefined);
                       setOpen(false);
                     }}
                     initialFocus
                   />
                 </PopoverContent>
               </Popover>
 
-              <Select>
+              <Select
+                value={formik.values.priority}
+                onValueChange={(value) =>
+                  formik.setFieldValue("priority", value as Priority)
+                }
+              >
                 <SelectTrigger className="w-[150px]">
                   <SelectValue placeholder="Priority" />
                 </SelectTrigger>
                 <SelectContent>
-                  <SelectItem value="hight">High</SelectItem>
+                  <SelectItem value="high">High</SelectItem>
                   <SelectItem value="medium">Medium</SelectItem>
                   <SelectItem value="low">Low</SelectItem>
                 </SelectContent>
               </Select>
             </div>
+          </div>
 
-            <div className="flex gap-1">
-              <Button
-                className="bg-gray-500 text-[14px] py-1 px-3 font-semibold rounded-sm cursor-pointer hover:bg-gray-400"
-                onClick={handleAddTodoClick}
-              >
-                Cancel
-              </Button>
+          <div className="flex gap-1 justify-end py-2 px-2">
+            <Button
+              type="button"
+              onClick={() => {
+                formik.resetForm();
+                handleAddTodoClick();
+              }}
+              className="bg-gray-500 text-[14px] py-1 px-3 font-semibold rounded-sm cursor-pointer hover:bg-gray-400"
+            >
+              Cancel
+            </Button>
 
-              <Button className="bg-primaryColor text-white text-[14px] py-1 px-3 font-semibold rounded-sm cursor-pointer hover:bg-[#a81f00bd]">
-                Add task
-              </Button>
-            </div>
+            <Button
+              type="submit"
+              className="bg-primaryColor text-white text-[14px] py-1 px-3 font-semibold rounded-sm cursor-pointer hover:bg-[#a81f00bd]"
+              disabled={!formik.values.text.trim() || formik.isSubmitting}
```

Comment:
Nit: Can we use `isValid` option here, if feasible?
```suggestion
              disabled={!formik.isValid || formik.isSubmitting}
```

Reply:
Can we keep `!formik.values.text.trim()` only? I am disabling the button when the text field is empty. The user can add a to-do item using only the text field, and other fields are optional if they choose not to fill them in.

Reply:
Okay, no problem.